### PR TITLE
Daskify and test avhrr_l1b_aapp  reader

### LINKED ISSUE #809: Update avhrr_l1b_aapp reader
**Describe the bug**
The avhrr_l1b_aapp reader hasn't been updated to be fully dask compliant and lacks unittests.

This is a reminder from #805

### PR DESCRIPTION
**Pull request by mraspaud**
This PR daskifies and tests the avhrr_l1b_aapp reader

Informal tests show a speedup from 7.9s to 5.5s to generate a composite on my laptop.

 - [x] Closes #809 
 - [x] Tests added and test suite added to parent suite 
 - [x] Tests passed 
 - [x] Passes ``flake8 satpy`` 
 - [x] Fully documented 
 - [x] Add your name to `AUTHORS.md` if not there already

### CODE REVIEW

**Comment by mraspaud** [comment]
@pnuu @adybbroe @djhoese does anyone want to review this or can I just merge ?

File: satpy/readers/aapp_l1b.py [hunk 1]
```diff
@@ -568,6 +570,5 @@ def _ir_calibrate(header, data, irchn, calib_type, mask=False):
         tb_ = (t_planck - bandcor_2) / bandcor_3
 
     # Mask unnaturally low values
-    # mask |= tb_ < 0.1
 
-    return np.where(mask, np.nan, tb_)
+    return da.where(mask, tb_, np.nan)
```
Comment (djhoese) [inline satpy/readers/aapp_l1b.py]: So the `mask` is the inverse of what it was?

Comment (mraspaud) [reply satpy/readers/aapp_l1b.py]: Yes it's been inversed

**Comment by mraspaud** [comment]
@djhoese @pnuu thanks for taking the time to review! Merging now.

**Pull request merged**